invenio_oauthclient/receivers.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+from flask import current_app
+from invenio_base.signals import secret_key_changed
+from invenio_db import db
+from invenio_oauthclient.models import RemoteToken
+from sqlalchemy import update
+from sqlalchemy_utils.types.encrypted import EncryptedType
+
+
+@secret_key_changed.connect
+def notify_secret_key_changed(sender, old_key):
+    """Rebuild the access_token field when the SECRET_KEY is changed.
+    
+    Needed to fix the access tokens used in the login.
+
+    :param sender: signal sender object.
+    :param old_key: the old SECRET_KEY.
+    """
+    print('invenio-oauthclient: changing SECRET_KEY...')
+    db.session.close_all()
+    eng = db.get_engine(current_app)
+    conn = eng.connect()
+    query_result = conn.execute("""select id_remote_account, access_token 
+                                   from oauthclient_remotetoken;""")
```

Comment:
That is what I tried at first but it results in a decoding error, as the model will try to decrypt the access_token using the new SECRET_KEY from app.config.